Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -311,6 +306,18 @@ private func createResolvedPackages(
             }
         }
     }
+    for targetName in allTargetNames.sorted() {
```

Comment:
I think we should create a new variable "foundDuplicateTarget" which gets set in the previous loop at around Line 276 in this diff. Then here we just check if its set and then find all the duplicate targets.